pkg/systemd/generate/common.go

```diff
@@ -60,13 +60,21 @@ func filterPodFlags(command []string) []string {
 	return processed
 }
 
-// quoteArguments makes sure that all arguments with at least one whitespace
+// escapeSystemdArguments makes sure that all arguments with at least one whitespace
 // are quoted to make sure those are interpreted as one argument instead of
-// multiple ones.
-func quoteArguments(command []string) []string {
+// multiple ones. Also make sure to escape all characters which have a special
+// meaning to systemd -> $,% and \
+// see: https://www.freedesktop.org/software/systemd/man/systemd.service.html#Command%20lines
+func escapeSystemdArguments(command []string) []string {
 	for i := range command {
+		command[i] = strings.ReplaceAll(command[i], "$", "$$")
+		command[i] = strings.ReplaceAll(command[i], "%", "%%")
 		if strings.ContainsAny(command[i], " \t") {
 			command[i] = strconv.Quote(command[i])
+		} else if strings.Contains(command[i], `\`) {
```

Comment:
I am completely lost here. How can this possibly work as an `else`?

Reply:
OK I think I understand: `strconv.Quote()` seems to do backslash interpretation. If this is the case, could you add one more test case, one in which the test string contains no spaces or tabs? Thank you!

Reply:
There is a test case at line 74-75 in common_test.go.

Reply:
My bad: I missed that, sorry.

One more question: will there ever be a condition in which we need to pass literals like `\n`, `\a`, `\007` ?

Reply:
I cannot think of any cases where this would be wanted.